zerver/management/commands/soft_activate_deactivate_users.py

```diff
@@ -0,0 +1,116 @@
+from argparse import ArgumentParser
+from six.moves import map
+import sys
+
+from zerver.models import UserProfile, UserMessage, Realm, RealmAuditLog, get_realm
+from zerver.lib.actions import get_users_for_soft_deactivation
+from zerver.lib.message import maybe_catch_up_soft_deactivated_user
+
+class Command(BaseCommand):
+    help = """Soft activate/deactivate users. Users are recognised by there emails here."""
+
+    def add_arguments(self, parser):
+        # type: (ArgumentParser) -> None
+        parser.add_argument('-r', '--realm',
+                            dest='string_id',
+                            type=str,
+                            required=True,
+                            help='The subdomain or string_id of the realm.')
+        parser.add_argument('-d', '--deactivate',
+                            dest='deactivate',
+                            action='store_true',
+                            default=False,
+                            help='Used to deactivate user/users.')
+        parser.add_argument('-a', '--activate',
+                            dest='activate',
+                            action='store_true',
+                            default=False,
+                            help='Used to activate user/users.')
+        parser.add_argument('--inactive-for',
+                            type=int,
+                            default=28,
+                            help='Specify the number of days of user inactivity that user should be marked soft_deactviated')
+        parser.add_argument('users', metavar='<users>', type=str, nargs='*', default=[],
+                            help="This option can be used to specify a list of user emails to soft activate/deactivate.")
+
+    def soft_deactivate_users(self, user_ids, realm):
+        # type: (List[int], Realm) -> None
+        cursor = connection.cursor()
+
+        soft_deactivation_query = '''
+        UPDATE zerver_userprofile SET
+            (long_term_idle, last_active_message_id) = (utm.long_term_idle, utm.last_active_message_id)
+        FROM
+            (SELECT
+                user_profile_id, max(message_id) AS last_active_message_id, True AS long_term_idle
+            FROM zerver_usermessage
+            WHERE
+                user_profile_id IN (%s)
+            GROUP BY user_profile_id) AS utm
+        WHERE
+            utm.user_profile_id = zerver_userprofile.id
+        ''' % (','.join(map(str, user_ids)))
+
+        cursor.execute(soft_deactivation_query)
+        realm_logs = []
+        event_time = timezone_now()
+        for id in user_ids:
+            log = RealmAuditLog(
+                realm=realm,
+                modified_user_id=id,
+                event_type='user_soft_deactivated',
+                event_time=event_time
+            )
+            realm_logs.append(log)
+
+        RealmAuditLog.objects.bulk_create(realm_logs)
+
+        print("Users deactivated: Total affected (%s)" % (cursor.rowcount))
```

Comment:
yeah, I actually dropped to the raw sql because I couldn't find a way of bulk updating in Django ORM. Basically they don't allow joins in updates. 
Idea was same to get a performance boost here.

Though post your thoughts I have refactored commits to extract out a new lib file `zerver/lib/soft_deactivation.py` which will be doing all this stuff for both the tests and management commands and anywhere else we might need it.